src/core/simulation_problems.jl

```diff
@@ -0,0 +1,67 @@
+"""
+Stores the OperationProblem definitions to be used in the simulation. When creating the
+SimulationProblems, the order in which the problems are created determines the order on which
+the simulation is executed.
+"""
+mutable struct SimulationProblems
+    op_problems::OrderedDict{Symbol, OperationsProblem}
+    names::Vector{Symbol}
+    function SimulationProblems(; kwargs...)
+        prob_dict = OrderedDict(kwargs...)
+        new(prob_dict, collect(keys(prob_dict)))
+    end
+end
+
+Base.getindex(problems::SimulationProblems, key::Symbol) = problems.op_problems[key]
+Base.getindex(problems::SimulationProblems, key) = getindex(problems, Symbol(key))
+
+Base.length(problems::SimulationProblems) = length(problems.op_problems)
+Base.first(problems::SimulationProblems) = first(problems.op_problems)
+Base.iterate(problems::SimulationProblems, args...) = iterate(problems.op_problems, args...)
+
+get_problem_names(problems::SimulationProblems) = problems.names
+get_problem_instances(problems::SimulationProblems) = values(problems.op_problems)
+
+function get_problem_numer(problems::SimulationProblems, name)
+    return findfirst(x -> x == Symbol(name), get_problem_names(problems))
+end
```

Comment:
This doesn't seem correct. I don't see anything about a number.

Reply:
find first returns the location of the function. This gets me the location in the array of names (which is ordered) where the name is.

Reply:
Is `index` a better name than `number`? To me `get_problem_index(...)` seems more appropriate.